src/main/java/book/shop/bookstore/mapper/BookMapper.java

```diff
@@ -29,7 +28,7 @@ default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
         bookDto.setCategoryIds(book.getCategories()
                 .stream()
                 .map(Category::getId)
-                .collect(Collectors.toSet()));
+                .toList());
     }
 
     @Named("bookFromId")
```

Comment:
This method is not mapping, it's copy of BookService.getBookById()
Please use service where necessary